### ISSUE TITLE
MIDI player "crashing" upon changing time slider

### ISSUE THREAD

**Issue by DBerke** [frescobaldi/frescobaldi#1295]
My apologies if this has already been raised, I skimmed all the issues with MIDI in the title and couldn't find anything about it. I turned on Frescobaldi's MIDI player and have been using that to listen to the music I'm working on, but quite frequently I run into the following bug:

```Traceback (most recent call last):
  File "/usr/share/frescobaldi/frescobaldi_app/miditool/widget.py", line 189, in slotTimeSliderChanged
    self._player.seek(value)
  File "/usr/share/frescobaldi/frescobaldi_app/midifile/player.py", line 154, in seek
    self.set_position(pos, offset)
  File "/usr/share/frescobaldi/frescobaldi_app/midifile/player.py", line 185, in set_position
    self.position_event(old, self._position)
  File "/usr/share/frescobaldi/frescobaldi_app/miditool/player.py", line 61, in position_event
    output.send_events(midi)
  File "/usr/share/frescobaldi/frescobaldi_app/midifile/output.py", line 132, in send_events
    self.output.write(l)
  File "/usr/share/frescobaldi/frescobaldi_app/portmidi/__init__.py", line 204, in write
    self._output.Write(data)
  File "/usr/share/frescobaldi/frescobaldi_app/portmidi/ctypes_pypm.py", line 112, in Write
    _check_error(err)
  File "/usr/share/frescobaldi/frescobaldi_app/portmidi/ctypes_pypm.py", line 188, in _check_error
    err_msg.decode('utf-8')))
portmidi.MidiException: PortMIDI-ctypes error [-10000]: Operation not permitted
```
This happens upon releasing the time slider on the MIDI player after dragging it to a new position, but there doesn't seem to be any pattern to it that I can detect; sometimes it happens the first time I try it after starting Frescobaldi, sometimes it only happens after dozens of times moving the slider. Once it happens the MIDI player refuses to play music again until I quit and restart Frescobaldi. It's not a huge problem, but it interrupts my workflow.

It's possible it might be a problem of me doing something I shouldn't (I'm only two months in to using Frescobaldi so I'm still pretty new), in which case I'd be very happy to be instructed as to how to make it stop!

**Comment by fedelibre** [comment]
"Operation not permitted"... are you sure your user have access to the soundcard device? Which Linux distro are you using?

I've just added a [Troubleshooting section in the wiki page about MIDI](https://github.com/frescobaldi/frescobaldi/wiki/MIDI-playback-on-Linux#troubleshooting), as I've just found out that in Fedora 32 the user must now be a member of the audio group in order to get access to the soundcard.

**Comment by DBerke** [comment]
Thanks for the reply. I'm running Debian 10 stable, "Buster". Running `groups` shows that `audio` is included for my user account:
```
$ groups
daniel cdrom floppy sudo audio dip video plugdev netdev bluetooth lpadmin scanner
```
Like I mentioned, the odd thing about this error is that it's intermittent. I can move the slider successfully perhaps 70–80% of the time, and hear music just fine. If there's some sort of pattern to when it fails I haven't been able to work it out yet.

**Comment by fedelibre** [comment]
Sorry, I didn't read carefully your report.
I've now tried playing with the slider but I never get any segfault. It might depend on different versions we are using of either Frescobaldi or some dependencies.

Here's what I'm testing now:

```
Frescobaldi: 3.1.2
Extension API: 0.9.0
Python: 3.8.2
python-ly: 0.9.6
Qt: 5.13.2
PyQt: 5.13.2
sip: 4.19.21
poppler: 0.84.0
python-poppler-qt: 0.75.0
OS: Linux-5.6.8-300.fc32.x86_64-x86_64-with-glibc2.2.5
```

```
$ dnf info python3-portmidi | grep Version
Version      : 217
```

Can you post here yours?

**Comment by DBerke** [comment]
Sure, no problem:
```
Frescobaldi: 3.0.0
Python: 3.7.3
python-ly: 0.9.5
Qt: 5.11.3
PyQt: 5.11.3
sip: 4.19.14
poppler: 0.63.0
python-poppler-qt: 0.24.2
OS: Linux-4.19.0-8-amd64-x86_64-with-debian-10.3
```
Apparently on Debian instead of `python3-portmidi` it's `python-pypm`, which is version 0.0.7-8.

This of course immediately suggests that the problem is due to having an older version of one of these things, and that it's been fixed in an updated version (which is nice to know!). And unfortunately the one downside of Debian stable's stability is a lot of difficulty getting newer software updates, so I suppose I'll just live with it for a few months till the next major update comes around. I probably should've thought to check the versions earlier, though. Thanks. I imagine this issue can be closed now.

**Comment by fedelibre** [comment]
> Apparently on Debian instead of `python3-portmidi` it's `python-pypm`, which is version 0.0.7-8.

I don't think it's the package actually used by Frescobaldi, as you can see in the [list of dependencies](https://packages.debian.org/buster/frescobaldi).

Anyway, as you are using 3.0.0, I think the problem was in Frescobaldi and was fixed in version 3.1. See #895 .
You have two possible solution: pin the version in testing (currently 3.1.1) or run Frescobaldi from source (as explained in the wiki).

I'm closing this issue.
